File: c/RansomNote.c
// https://leetcode.com/problems/ransom-note/

bool canConstruct(char* ransomNote, char* magazine) {
    int i, table[26] = {0};
    
    while(*magazine) {
        table[*magazine - 'a']++;
        magazine++;
    }
    
    while(*ransomNote) {
        table[*ransomNote - 'a']--;
        ransomNote++;
    }
    
    for(i = 0; i < 26; i++)
        if(table[i] < 0)
            return false;
    
    return true;
}

File: swift/RansomNote.swift
// https://leetcode.com/problems/ransom-note/

class Solution {
    func canConstruct(_ ransomNote: String, _ magazine: String) -> Bool {
        var dict = [Character: Int]()
        magazine.forEach {
            dict[$0, default: 0] += 1
        }
        for i in ransomNote {
            if let count = dict[i], count > 0 {
                dict[i] = count - 1
            } else {
                return false
            }
        }
        return true
    }
}
